### PR TITLE
On over abstracting

### PR DESCRIPTION
**Pull request by hannesvdvreken**
Blog post on over-abstracting

@madewithlove/development

### CODE REVIEW

**Comment by hannesvdvreken** [comment]
Things I didn't cover but will cover in later blog posts:

- Interface segregation
- Everything should either be an interface, class (final implementation) or trait (partial implementation) - nothing in between.

**Comment by mogetutu** [comment]
:+1:

File: _posts/2015-09-25-on-over-abstracting.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        }
+
+        return $this->models->get();
+    }
+}
+```
+
+Clearly what's happening here is: the base class had a `get` method which returns a list of models. As more use cases appear, the base class doesn't have enough flexibility. The `get` method is being changed to take an optional argument `$options`. As the amount of possible use cases grows, the `get` method's LOC (lines of code) grows as well.
+
+In the end the `get` method is chuck full of bugs, too hard to test and too hard to adapt without breaking it. Developers that need slightly different behavior for their AbstractRepository implementation will eventually override the `get` method with their own simple method. Result: a simpler method with less features, and thus: loss of reusable functionality.
```
Comment (tonysm) [inline _posts/2015-09-25-on-over-abstracting.md]: `is chuck full of bugs` ~> `is a chunk full of bugs`?

Comment (hannesvdvreken) [inline _posts/2015-09-25-on-over-abstracting.md]: Should be `chock-full`.

**Comment by tonysm** [comment]
:+1:

**Pull request merged**